src/test/optimizer/strategy/predicate_merge_rule_test.cpp

```diff
@@ -0,0 +1,278 @@
+#include "strategy_base_test.hpp"
+
+#include "expression/expression_functional.hpp"
+#include "logical_query_plan/mock_node.hpp"
+#include "logical_query_plan/predicate_node.hpp"
+#include "logical_query_plan/projection_node.hpp"
+#include "logical_query_plan/union_node.hpp"
+#include "optimizer/strategy/predicate_merge_rule.hpp"
+#include "testing_assert.hpp"
+
+using namespace opossum::expression_functional;  // NOLINT
+
+namespace opossum {
+
+class PredicateMergeRuleTest : public StrategyBaseTest {
+ public:
+  void SetUp() override {
+    node_a = MockNode::make(MockNode::ColumnDefinitions{{DataType::Int, "a"}, {DataType::Int, "b"}});
+    a_a = node_a->get_column("a");
+    a_b = node_a->get_column("b");
+
+    node_b = MockNode::make(MockNode::ColumnDefinitions{{DataType::Int, "a"}, {DataType::Int, "b"}}, "b");
+    b_a = node_b->get_column("a");
+    b_b = node_b->get_column("b");
+
+    rule = std::make_shared<PredicateMergeRule>();
+    // A value of 1 means that the rule won't merge pure predicate chains without UNIONs.
```

Comment:
Rather: "Reducing the optimization_threshold so that plans are merged earlier, making the test cases shorter"?